api/controllers/DashboardController.js

```diff
@@ -144,48 +144,30 @@ const dashboardController = {
                 sails.log.error('No results found.');
             }
         }
-        const apiResponse = await dashboardController._getDataFromCasebook(
-            req,
-            results
-        );
+        const { data: apiResponse } =
+            await dashboardController._getDataFromCasebook(results);
+
         return dashboardController._addCasebookStatusesToResults(apiResponse, {
             ...displayAppsArgs,
             results,
         });
     },
 
-    _getDataFromCasebook(req, results) {
-        const applicationReferences = results.map(
-            (resultItem) => resultItem.unique_app_id
-        );
-
-        const queryParamsObj = {
-            timestamp: Date.now().toString(),
-            applicationReference: applicationReferences,
-        };
+    async _getDataFromCasebook(results) {
+        try {
+            const applicationReferences = results.map(
+                (resultItem) => resultItem.unique_app_id
+            );
+            const queryParamsObj = {
+                timestamp: Date.now().toString(),
+                applicationReference: applicationReferences,
+            };
```

Comment:
this looks duplicated in `FileDownloadController` as well. 

it seems like to get an application status, you always have to pass a timestamp, as well as application reference(s). `CasebookService.getApplicationStatus` can handle adding timestamp itself. 

```
//(just shorthand, don't add it as an anon function)
CasebookService.getApplicationStatus = (references) => ({

const query = { 
  timestamp: Date.now(),
  applicationReference: references
}
 //..
})
```

Reply:
Good point. I'm sure there was a time where these objects were different, but maybe I refactored that away 🤷

Reply:
🪄